embedded-google-storage/src/main/java/com/playtika/test/storage/EmbeddedStorageBootstrapConfiguration.java

```diff
@@ -0,0 +1,117 @@
+package com.playtika.test.storage;
+
+import com.playtika.test.common.spring.DockerPresenceBootstrapConfiguration;
+import com.playtika.test.common.utils.ContainerUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.shaded.okhttp3.MediaType;
+import org.testcontainers.shaded.okhttp3.OkHttpClient;
+import org.testcontainers.shaded.okhttp3.Request;
+import org.testcontainers.shaded.okhttp3.RequestBody;
+import org.testcontainers.shaded.okhttp3.Response;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+
+import static com.playtika.test.common.utils.ContainerUtils.configureCommonsAndStart;
+import static java.lang.String.format;
+
+@Slf4j
+@Configuration
+@ConditionalOnExpression("${embedded.containers.enabled:true}")
+@AutoConfigureAfter(DockerPresenceBootstrapConfiguration.class)
+@ConditionalOnProperty(name = "embedded.google.storage.enabled", matchIfMissing = true)
+@EnableConfigurationProperties(StorageProperties.class)
+public class EmbeddedStorageBootstrapConfiguration {
+
+    @Bean(name = StorageProperties.BEAN_NAME_EMBEDDED_GOOGLE_STORAGE_SERVER, destroyMethod = "stop")
+    GenericContainer<?> storageServer(
+        ConfigurableEnvironment environment,
+        StorageProperties properties) throws IOException {
+        GenericContainer<?> container = new GenericContainer<>(ContainerUtils.getDockerImageName(properties))
+            .withExposedPorts(StorageProperties.PORT)
+            .withCreateContainerCmdModifier(cmd -> cmd.withEntrypoint(
+                "/bin/fake-gcs-server",
```

Comment:
shouldnt we also provide `-backend memory` option? is is provided in examples and i don't see what is the default value for it.